src/core/cwa.ts

```diff
@@ -32,6 +46,40 @@ export default class Cwa {
     this.$state = storage.state
   }
 
+  private getRouteName (path) {
+    return path.split('/').reduce((acc, urlPart, i, urlParts) => i === urlParts.length - 1 ? acc + '' : acc + urlPart.replace('_', ''), '')
```

Comment:
Not sure about this, mb that it should be implicit instead in `fetchRoute` with a `name` argument?

Reply:
I think that would be safer and simpler. I'm also considering whether all components should have the name 'component' instead of each individual one. I'll have a think about the use case this week and whether it'll be important to be able to query all of these. I'm even thinking if it'd possibly end up having a structure for the components similar to:
```
{
  "components": {
    "html_component": {
       // same structure as already existing here
    },
    "another_component": {
      // etc...
    }
    "allIds": [ "/component/html_components/1234", "/component/another_components/1234" ]
  }
}
```

Will have a think...

Reply:
Currently I don't think this is needed so I'll resolve the conversation - no changes.